client/my-sites/checkout/controller.jsx

```diff
@@ -72,6 +72,14 @@ export function checkout( context, next ) {
 		return;
 	}
 
+	const isComingFromSignup = !! context.query.signup;
+	if ( isComingFromSignup ) {
+		const currentHref = document.location.href;
+		window.history.pushState( null, null, `/plans/${ selectedSite?.slug ?? '' }` );
+		window.history.pushState( null, null, `/plans/${ selectedSite?.slug ?? '' }` );
```

Comment:
Is this line supposed to be duplicated?

Reply:
yes, we add two entries then call replace state so the current url doesn't point to /plans